Update Lighthouse Book for Electra features (#7280)
sigp/lighthouse: book/src/developers_architecture.md

# Lighthouse architecture

> A technical walkthrough on Lighthouse based on a Lighthouse architecture below can be found at: [Lighthouse technical walkthrough](https://www.youtube.com/watch?v=pLHhTh_vGZ0)

Reviewer: ```suggestion
A technical walkthrough of Lighthouse's architecture can be found at: [Lighthouse technical walkthrough](https://www.youtube.com/watch?v=pLHhTh_vGZ0)
```
